src/csharp/Microsoft.Spark.Worker/Command/SqlCommandExecutor.cs

```diff
@@ -737,6 +738,25 @@ internal ArrowOrDataFrameGroupedMapCommandExecutor(Version version)
             return ExecuteArrowGroupedMapCommand(inputStream, outputStream, commands);
         }
 
+        private RecordBatch WrapArrowRecordBatchColumnsInAStruct(RecordBatch batch)
+        {
+            if (SparkEnvironment.SparkVersion >= new Version(Versions.V3_0_0))
+            {
+                ArrowBuffer.BitmapBuilder validityBitmapBuilder = new ArrowBuffer.BitmapBuilder();
+                for (int i = 0; i < batch.Length; i++)
+                {
+                    validityBitmapBuilder.Append(true);
+                }
+                ArrowBuffer validityBitmap = validityBitmapBuilder.Build();
+
+                StructType structType = new StructType(batch.Schema.Fields.Select((KeyValuePair<string, Field> pair) => pair.Value).ToList());
```

Comment:
Can we not use Linq here? It feels like it isn't worth the extra runtime cost.

Reply:
yup yup, was going to clean that up on the next pass